architecture/2025-04-aci-confidential-containers.md

```diff
@@ -0,0 +1,388 @@
+Confidential containers require a CCE policy to be set on the container group profile associated with ACI instances. CCE policies can only be generated using an extension on the Azure CLI. As a result, Radius will need access to the Azure CLI and the `confcom` extension to support this feature. More information on confidential containers with ACI can be found [here](https://learn.microsoft.com/en-us/azure/container-instances/container-instances-tutorial-deploy-confidential-containers-cce-arm)
+
+The key components that will be deployed by Radius to support ACI are container group profiles, NGroups, and the container instance itself. 
+
+## Terms and definitions
+
+- ACI: Azure Container Instances
+- TEE: Trusted Execution Environment
+- Confidential Containers: Containers that run in a hardware-backed TEE to provide enhanced security features
+- Container Group Profile: A configuration template in Azure Container Instances that defines the properties and settings for a container group. It includes specifications such as container images, resource allocations (CPU/memory), networking settings, storage mounts, OS type, SKU (Standard/Dedicated/Confidential), and security policies. For confidential containers, the Container Group Profile contains the CCE policy that enables the Trusted Execution Environment.
+
+## Objectives
+
+This document is one component of the larger serverless extensibility experience. See below for relevant design documents. 
+
+### Goals
+
+- Support the deployment of application containers to ACI, ensuring that container definitions are platform-agnostic.
+- Allow users to apply ACI-specific configurations to containers using extension and runtime properties.
+- Support generating a CCE policy as part of a container group profile creation. 
+
+### Non-goals
+
+- A serverless extensibility design. This document will only focus on components ot Radius relevant to confidential container deployment. Compute extensibility design can be found [here](https://github.com/radius-project/design-notes/pull/91).  
+- Multi-compute support per environment will not be available initially.  
+- Full Dapr parity will not be available initially.  
+- ACI Service Discovery.
+
+### User scenarios (optional)
+
+<!--
+Describe the user scenarios for this design. Ensure that you define the
+roles and personas in these user scenarios when it requires API design.
+If you have an existing issue that describes the user scenarios, please
+link to that issue instead.
+-->
+
+The scenario document for severless is in progress [here](https://github.com/radius-project/design-notes/blob/9bcaa014c013b254593426280add5ef5c69b265e/architecture/2025-01-serverless-feature-spec.md#scenario-3-punch-through-to-platform-specific-features-and-incremental-adoption-of-radius-into-existing-serverless-applications). Please refer to the scenarios defined in that doc, specifically scenario 3 as it relates to `runtimes` punchthroughs. 
+
+## User Experience (if applicable)
```

Comment:
Alternative option: require the user to have the azure CLI + `confcon` extension downloaded and accessible to Radius before using CC on ACI

Reply:
Updated to use this option